packages/flutter/lib/src/rendering/image_filter_config.dart

```diff
@@ -0,0 +1,174 @@
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+/// A configuration for a [ui.ImageFilter].
+///
+/// This class provides a framework-level abstraction for image filters that can
+/// be applied to widgets or render objects. Unlike [ui.ImageFilter], which is an
+/// engine-level object, [ImageFilterConfig] instances are created at the widget
+/// or render object level and can be resolved into a [ui.ImageFilter] at layout
+/// time, allowing them to incorporate layout-dependent information such as the
+/// widget's bounds.
+///
+/// Most filters can be used via [ImageFilterConfig.filter]. The most notable
+/// filter that requires this class is [ImageFilterConfig.blur] with the
+/// `bounded` option set to true.
+///
+/// See also:
+///
+///  * [BackdropFilter.filterConfig], which uses this class to configure its filter.
+///  * [ui.ImageFilter], the engine-level class that this config resolves to.
+@immutable
+abstract class ImageFilterConfig {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const ImageFilterConfig();
+
+  /// Creates a configuration that directly uses the given filter.
+  factory ImageFilterConfig.fromImageFilter(ui.ImageFilter filter) {
+    return _DirectImageFilterConfig(filter);
+  }
+
+  /// Creates a configuration for a Gaussian blur.
+  ///
+  /// The `sigmaX` and `sigmaY` arguments are the standard deviation of the
+  /// Gaussian kernel in the x and y directions, respectively.
+  ///
+  /// The `tileMode` argument determines how the blur should handle edges.
+  ///
+  /// If `bounded` is true, the filter performs a "bounded blur". This means the
+  /// blur kernel will only sample pixels from within the provided attached
+  /// render object, treating all pixels outside of it as transparent. This
+  /// mode is typically used to implement high-fidelity iOS-style blurs.
+  factory ImageFilterConfig.blur({
+    double sigmaX = 0.0,
+    double sigmaY = 0.0,
+    ui.TileMode tileMode = ui.TileMode.clamp,
+    bool bounded = false,
+  }) {
+    return _BlurImageFilterConfig(
+      sigmaX: sigmaX,
+      sigmaY: sigmaY,
+      tileMode: tileMode,
+      bounded: bounded,
+    );
+  }
+
+  /// Composes the `inner` filter configuration with `outer`, to combine their
+  /// effects.
+  ///
+  /// Creates a single [ImageFilterConfig] that when applied, has the same
+  /// effect as subsequently applying `inner` and `outer`, i.e., result =
+  /// outer(inner(source)).
+  factory ImageFilterConfig.compose({
+    required ImageFilterConfig outer,
+    required ImageFilterConfig inner,
+  }) {
+    return _ComposeImageFilterConfig(outer: outer, inner: inner);
+  }
+
+  /// Resolves this configuration into a [ui.ImageFilter], given the
+  /// `bounds` of the widget applying the filter.
+  ///
+  /// The `bounds` can be used to create layout-dependent filters, such as
```

Comment:
somewhere in the doc should mention what coordinate this bound is in. based on the code, This is in global coordinate

Reply:
Fixed.

Reply:
Update: Since the parameter here has been changed to a `ImageFilterContext`, it no longer need to define the bounds. Instead, `ImageFilterContext.bounds` contains a complete definition of the bounds, including the coordinate it uses.